core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -710,6 +726,38 @@ private void maybeTrimTranslog() {
         }
     }
 
+    private void syncGlobalCheckpoints() {
+        for (final IndexShard shard : this.shards.values()) {
+            if (shard.routingEntry().active() && shard.routingEntry().primary()) {
+                switch (shard.state()) {
+                    case CREATED:
+                    case RECOVERING:
+                    case CLOSED:
+                        continue;
+                    case POST_RECOVERY:
```

Comment:
routing entry cannot be active while shard state is POST_RECOVERY ;-)

Reply:
Addressed in 9bf875e86115c72f5cbd2d03e835ded9a04f9ff9.